### ISSUE TITLE
FTP Error when try to download events folder

### ISSUE THREAD

**Issue by piMyHome** [ctlcltd/e2-sat-editor#14]
### Describe the bug

When clicking the download button, files are getting downloaded normally until  e2-sat-editor tries to download the folder
/etc/enigma2/events which results in error:
FTP Error: Remote file not found

### Steps to reproduce

I'm not sure if you cane reproduce it, but I added all logs so it should be clear that the issue is the folder events.

### Expected behavior

Maybe ignoring the folder events when downloading. I do not even know why e2-sat-editor needs the bash scripts in events folder

### Software version

1.1

### Download medium

GitHub Releases page

### Platform running

Linux Gnome

### Architecture running

x86_64 (64-bit)

### Other (please specify)

Vu+ Duo 4K 
e2-sat-editor-1.1.0-x86_64-linux.AppImage
Ubuntu 22.04.3 LTS jammy

### Log

```shell
# -----------------
# GUI Log Inspector
# -----------------
2024-01-20 12:56:51.701436 <Debug> [e2se.gui] tab::ftpDownload
2024-01-20 12:56:51.701620 <Debug> [e2se.gui] ftpcom::connect
2024-01-20 12:56:51.714981 <Debug> [e2se.gui] ftpcom::fetch_paths
2024-01-20 12:56:51.714994 <Debug> [e2se.gui] ftpcom::list_dir
2024-01-20 12:56:51.719322 <Debug> [e2se.gui] ftpcom::list_dir
2024-01-20 12:56:51.721305 <Debug> [e2se.gui] ftpcom::download_data
2024-01-20 12:56:51.721309 <Debug> [e2se.gui] ftpcom::download_data  file: /etc/enigma2/blacklist
2024-01-20 12:56:51.725193 <Debug> [e2se.gui] ftpcom::download_data
2024-01-20 12:56:51.725196 <Debug> [e2se.gui] ftpcom::download_data  file: /etc/enigma2/bouquets.radio
2024-01-20 12:56:51.726648 <Debug> [e2se.gui] ftpcom::download_data
2024-01-20 12:56:51.726651 <Debug> [e2se.gui] ftpcom::download_data  file: /etc/enigma2/bouquets.tv
2024-01-20 12:56:51.731260 <Debug> [e2se.gui] ftpcom::download_data
2024-01-20 12:56:51.731266 <Debug> [e2se.gui] ftpcom::download_data  file: /etc/enigma2/ci1.xml
2024-01-20 12:56:51.733662 <Debug> [e2se.gui] ftpcom::download_data
2024-01-20 12:56:51.733666 <Debug> [e2se.gui] ftpcom::download_data  file: /etc/enigma2/dvbapp2debug.conf
2024-01-20 12:56:51.735265 <Debug> [e2se.gui] ftpcom::download_data
2024-01-20 12:56:51.735269 <Debug> [e2se.gui] ftpcom::download_data  file: /etc/enigma2/epgrefresh.xml
2024-01-20 12:56:51.737645 <Debug> [e2se.gui] ftpcom::download_data
2024-01-20 12:56:51.737648 <Debug> [e2se.gui] ftpcom::download_data  file: /etc/enigma2/events                 # events is a directory, not a file!
2024-01-20 12:56:51.738420 <Error> [e2se.gui] ftpcom::download_data  FTP Error: Remote file not found

# ------------------------------------------------------
# Terminal Log e2-sat-editor-1.1.0-x86_64-linux.AppImage 
# ------------------------------------------------------
* Connecting to 192.168.60.38 (192.168.60.38) port 48379
*   Trying 192.168.60.38:48379...
* Connected to 192.168.60.38 (192.168.60.38) port 21
> SIZE epgrefresh.xml
< 213 2931
> RETR epgrefresh.xml
< 150 Opening BINARY mode data connection for epgrefresh.xml (2931 bytes).
* Maxdownload = -1
* Getting file with size: 2931
* Remembering we are in dir "/etc/enigma2/"
< 226 Transfer complete.
* Connection #1 to host 192.168.60.38 left intact
* Found bundle for host: 0x7fc744019d40 [serially]
* Re-using existing connection with host 192.168.60.38
* Request has same path as previous transfer
> EPSV
* Connect data stream passively
* ftp_perform ends with SECONDARY: 0
< 229 Entering Extended Passive Mode (|||12292|).
* Connecting to 192.168.60.38 (192.168.60.38) port 12292
*   Trying 192.168.60.38:12292...
* Connected to 192.168.60.38 (192.168.60.38) port 21
> SIZE events                                                                                # events is a directory, not a file!
< 550 Could not get file size.
* The file does not exist
* Remembering we are in dir "/etc/enigma2/"
* Connection #1 to host 192.168.60.38 left intact
2024-01-20 12:56:51.738420 <Error> [e2se.gui] ftpcom::download_data  FTP Error: Remote file not found

# ------------------------------
# vuduo4k:# cat /var/log/xferlog
# ------------------------------
Sat Jan 20 12:56:51 2024 1 192.168.60.47 0 /etc/enigma2/blacklist b _ o r root ftp 0 * c
Sat Jan 20 12:56:51 2024 1 192.168.60.47 119 /etc/enigma2/bouquets.radio b _ o r root ftp 0 * c
Sat Jan 20 12:56:51 2024 1 192.168.60.47 567 /etc/enigma2/bouquets.tv b _ o r root ftp 0 * c
Sat Jan 20 12:56:51 2024 1 192.168.60.47 226 /etc/enigma2/ci1.xml b _ o r root ftp 0 * c
Sat Jan 20 12:56:51 2024 1 192.168.60.47 5 /etc/enigma2/dvbapp2debug.conf b _ o r root ftp 0 * c
Sat Jan 20 12:56:51 2024 1 192.168.60.47 2931 /etc/enigma2/epgrefresh.xml b _ o r root ftp 0 * c

# -----------------------------------
vuduo4k:~# ls -l /etc/enigma2/events/
# -----------------------------------
-rwxrwxr-x    1 root     root           223 Jan  1  1970 DBTASK_CANCEL.sh
-rwxrwxr-x    1 root     root           223 Jan  1  1970 DBTASK_FINISH.sh
-rwxrwxr-x    1 root     root           223 Jan  1  1970 DBTASK_START.sh
-rwxrwxr-x    1 root     root           223 Jan  1  1970 E2START.sh
-rwxrwxr-x    1 root     root           223 Jan  1  1970 GUI_REBOOT.sh
-rwxrwxr-x    1 root     root           223 Jan  1  1970 PVRDESCRAMBLE_START.sh
-rwxrwxr-x    1 root     root           223 Jan  1  1970 PVRDESCRAMBLE_STOP.sh
-rwxrwxr-x    1 root     root           223 Jan  1  1970 REBOOT.sh
-rwxrwxr-x    1 root     root           223 Jan  1  1970 RECORD_REMIND.sh
-rwxrwxr-x    1 root     root           223 Jan  1  1970 RECORD_START.sh
-rwxrwxr-x    1 root     root           223 Jan  1  1970 RECORD_STOP.sh
-rwxrwxr-x    1 root     root           223 Jan  1  1970 RECORD_WAKEUP.sh
-rwxrwxr-x    1 root     root           223 Jan  1  1970 SERVICE_START.sh
-rwxrwxr-x    1 root     root           223 Jan  1  1970 SERVICE_STOP.sh
-rwxrwxr-x    1 root     root           223 Jan  1  1970 SHUTDOWN.sh
-rwxrwxr-x    1 root     root           223 Jan  1  1970 STANDBY_ENTER.sh
-rwxrwxr-x    1 root     root           223 Jan  1  1970 STANDBY_LEAVE.sh
-rwxrwxr-x    1 root     root           223 Jan  1  1970 STBBOOT.sh
-rwxrwxr-x    1 root     root           223 Jan  1  1970 TASK_CANCEL.sh
-rwxrwxr-x    1 root     root           223 Jan  1  1970 TASK_FINISH.sh
-rwxrwxr-x    1 root     root           223 Jan  1  1970 TASK_START.sh
```

**Comment by ctlcltd** [comment]
Thank you for reporting that @piMyHome. Same as #12 
There's a bug. A directory inside the paths interrupts the download. FTP should be more fault tolerant.
**I'm releasing a minor version 1.1.1, it should be available next 24-48 hours.**

**Comment by piMyHome** [comment]
Thanks @ctlcltd for your great work!

I was not sure if #12 has the same "directory" issue, since I didn't find anything similar in his post, so I opened a new ticket. Sorry for that....

**Comment by ctlcltd** [comment]
Really useful reporting, thank you @piMyHome !
If I hadn't seen the directory in #12, I wouldn't have understood the problem, your report would have been essential.
My STB has no directories in the paths (Transponders, Services, Bouquets), I would hardly have noticed the problem...
When I added the threads to FTP I put a stop after the first error, as result if there's an error it doesn't download anything.
Now in 1.1.1 it's more permissive

**Comment by ctlcltd** [comment]
Fixed on release [v1.1.1](../releases/tag/v1.1.1)
